client/src/client.rs

```diff
@@ -971,6 +1019,21 @@ impl<B, E, Block, RA> Client<B, E, Block, RA> where
 		Ok(())
 	}
 
+	fn notify_any_block_imported(&self, notify_import: &ImportSummary<Block>) -> sp_blockchain::Result<()> {
+
+		let notification = AllBlocksNotification::<Block> {
+			hash: notify_import.hash.clone(),
+			origin: notify_import.origin.clone(),
+			header: notify_import.header.clone(),
+			is_new_best: notify_import.is_new_best,
+		};
+
+		self.all_blocks_notification_sinks.lock()
+			.retain(|sink| sink.unbounded_send(notification.clone()).is_ok());
```

Comment:
At this stage I am wary of introducing new unbounded senders. @mxinden is already on a holy mission to remove all the existing ones. However, I don't see a way around this without executing this function asynchronously.

Reply:
That was precisely the motivation to start with the approach as is. See #5455 for more details / discussion on how to get rid of it, this is still ongoing.

Reply:
Removed.